packages/components/src/box-control/stories/index.story.tsx

```diff
@@ -0,0 +1,76 @@
+/**
+ * External dependencies
+ */
+import type { Meta, StoryFn } from '@storybook/react';
+
+/**
+ * WordPress dependencies
+ */
+import { useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import BoxControl from '../';
+import type { BoxControlValue } from '../types';
+
+const meta: Meta< typeof BoxControl > = {
+	title: 'Components (Experimental)/BoxControl',
+	component: BoxControl,
+	argTypes: {
+		onChange: { action: 'onChange' },
+	},
+	parameters: {
+		controls: { expanded: true },
+		docs: { canvas: { sourceState: 'shown' } },
+	},
+};
+export default meta;
+
+const defaultSideValues = {
+	top: '10px',
+	right: '10px',
+	bottom: '10px',
+	left: '10px',
+};
+
+const Template: StoryFn< typeof BoxControl > = ( props ) => {
+	const [ values, setValues ] =
+		useState< BoxControlValue >( defaultSideValues );
+
+	return (
+		<BoxControl
+			label="Padding"
+			values={ values }
+			{ ...props }
+			onChange={ ( nextValue ) => setValues( nextValue ) }
```

Comment:
Right now, any value of the `onChange` prop passed via `props` is overridden by the internal `onChange` implementation.

In order to make sure that the `onChange` prop gets called correctly, we have to do something like

```jsx
onChange={ ( nextValue ) => {
  setValues( nextValue );
  props.onChange?.( nextValue );
}
```